tests/framework/test_time.py

```diff
@@ -56,94 +62,123 @@ def step_column(sim):
+        self.ts_pipeline_value = None
 
     def setup(self, builder) -> None:
         super().setup(builder)
         builder.value.register_value_modifier("simulant_step_size", self.modify_step)
+        self.rate_pipeline = builder.value.register_value_producer(
+            "test_rate",
+            source=lambda idx: pd.Series(1.75, index=idx),
+            preferred_post_processor=rescale_post_processor,
+        )
 
-    def modify_step(self, index):
-        return pd.Series(pd.Timedelta(days=self.step_modifier), index=index)
-
+    def on_time_step(self, event: Event) -> None:
+        self.ts_pipeline_value = self.rate_pipeline(event.index)
 
-def test_align_times(SimulationContext, base_config, components):
+    def modify_step(self, index):
+        return pd.Series(
+            [
+                pd.Timedelta(days=self.step_modifier_even)
+                if i % 2 == 0
+                else pd.Timedelta(days=self.step_modifier_odd)
+                for i in index
+            ],
+            index=index,
+        )
+
+
+def test_basic_iteration(SimulationContext, base_config, components):
+    """Ensure that the basic iteration of the simulation works as expected.
+    The step size should always be 1 in this case, the whole population should
+    be updated, and the pipeline step value should always match the column step value.
+    """
+    base_config["time"]["step_size"] = 1
     sim = SimulationContext(base_config, components)
+    listener = [c for c in components if "listener" in c.args][0]
     sim.setup()
     sim.initialize_simulants()
+    assert sim._clock.time == get_time_stamp(sim.configuration.time.start)
+    assert sim._clock.step_size == pd.Timedelta(days=1)
     pop_size = len(sim.get_population())
     # After initialization, all simulants should be aligned to event times
     assert len(active_simulants(sim)) == pop_size
+    assert np.all(step_pipeline(sim) == step_column(sim))
 
-    sim.step()
-    # After one step (and no step adjustments, simulants should still be aligned)
-    assert len(active_simulants(sim)) == pop_size
-
-    sim._population._population.step_size *= 2
-
-    sim.step()
-    # No simulants should be aligned after a step size adjustment
-    assert active_simulants(sim).empty
-
-    sim.step()
-    # Now they should be aligned again
-    assert len(active_simulants(sim)) == pop_size
+    for _ in range(2):
+        taken_step_size = take_step(sim)
+        # After a step (and no step adjustments, simulants should still be aligned)
+        assert taken_step_size == pd.Timedelta(days=1)
+        assert len(active_simulants(sim)) == pop_size
+        assert np.all(step_pipeline(sim) == step_column(sim))
+        for index in listener.event_indexes.values():
+            assert np.all(index == sim.get_population().index)
 
 
-def test_unequal_steps(SimulationContext, base_config, components):
+def test_empty_active_pop(SimulationContext, base_config, components):
```

Comment:
I thought it was impossible to have an empty active population, since we'll always set the next time to the smallest next event time. That being said, we probably do need to test that the clock will still function if the population is empty or if there are no tracked simulants remaining in the population.

Reply:
Yeah, that's the intent behind this test--it shouldn't ever "naturally" be the case that we end up with an empty population, but I wanted to be sure that even in this case, we just take a minimum step size.